ITR/utils.py

```diff
@@ -124,3 +130,27 @@ def calculate(portfolio_data: pd.DataFrame, fallback_score: Quantity['delta_degC
         scores = ts.anonymize_data_dump(scores)
 
     return scores, aggregations
+
+
+# https://stackoverflow.com/a/74137209/1291237
+def umean(quantified_data):
+    """
+    Assuming Gaussian statistics, uncertainties stem from Gaussian parent distributions. In such a case,
+    it is standard to weight the measurements (nominal values) by the inverse variance.
+
+    This function uses error propagation on the to get an uncertainty of the weighted average.
+    :param: A set of uncertainty values
+    :return: The weighted mean of the values, with a freshly calculated error term
+    """
+    values = np.array(list(map(lambda v: v.m if isinstance(v.m, UFloat) else ufloat(v.m, 0),  quantified_data)))
+    epsilon = 1e-7
+    wavg = ufloat(sum([v.n/(v.s**2+epsilon) for v in values])/sum([1/(v.s**2+epsilon) for v in values]), 
+                  np.sqrt(len(values)/sum([1/(v.s**2+epsilon) for v in values])))
+    if wavg.s==0.0:
```

Comment:
The fact that you pushed a fix in a separate commit tells me, that we need a unit test coverage for this method

Reply:
Completely correct.  I will create an issue for creating the test case.  I will have some time on a plane today to write the test case to close the issue.

Reply:
i believe unit tests need to be added now.
otherwise we cannot guarantee the quality of this method